### PR TITLE
CWG Poll 11: P2082R1 Fixing CTAD for aggregates

### LINKED ISSUE #3695: [2020-02 CWG Poll 11] P2082R1 Fixing CTAD for aggregates
http://wiki.edg.com/pub/Wg21prague/StrawPolls/p2082r1.pdf

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #3695.
Fixes cplusplus/papers#809.

### CODE REVIEW

File: source/overloading.tex [hunk 1]
```diff
@@ -1571,23 +1571,49 @@
...
 of the \grammarterm{braced-init-list}, or
 of the \grammarterm{expression-list}.
-For each $x_i$, let $e_i$ be the corresponding element
+For each $x_i$, let $e_i$ be the corresponding aggregate element
 of \tcode{C} or of one of its (possibly recursive) subaggregates
-that would be initialized by $x_i$\iref{dcl.init.aggr}
-if brace elision is not considered for any element
-that has a dependent type.
-If there is no such element $e_i$, the program is ill-formed.
+that would be initialized by $x_i$\iref{dcl.init.aggr} if
```
Comment (burblebee) [inline source/overloading.tex]: The paper wants a ":" added after "if".

File: source/overloading.tex [hunk 2]
```diff
@@ -1571,23 +1571,49 @@
...
+\item
+a trailing aggregate element that is a pack expansion is assumed to correspond
+to all remaining elements of the initializer list (if any).
+\end{itemize}
+If there is no such aggregate element $e_i$ for any $x_i$,
+the aggregate deduction candidate is not added to the set.
 The aggregate deduction candidate is derived as above
 from a hypothetical constructor $\tcode{C}(\tcode{T}_1, \dotsc, \tcode{T}_n)$,
-where $\tcode{T}_i$ is the declared type of the element $e_i$.
+where
```
Comment (burblebee) [inline source/overloading.tex]: The paper wants a ":" appended here

File: source/overloading.tex [hunk 3]
```diff
@@ -1571,23 +1571,49 @@
...
+\begin{itemize}
+\item
+if $e_i$ is of array type and
+$x_i$ is a \grammarterm{braced-init-list} or string literal,
+$T_i$ is an rvalue reference to the declared type of $e_i$, and
```
Comment (burblebee) [inline source/overloading.tex]: T should not be in math font, use $\tcode{T}_i$ instead

File: source/overloading.tex [hunk 4]
```diff
@@ -1750,6 +1785,21 @@
...
+struct F : Types<T...>, T... {};
+
+struct X {};
+struct Y {};
+struct Z {};
+struct W { operator Y(); };
+
+F f1 = {Types<X, Y, Z>{}, {}, {}};      // OK, F<X, Y, Z> deduced
+F f2 = {Types<X, Y, Z>{}, X{}, Y{}};    // OK, F<X, Y, Z> deduced
+F f3 = {Types<X, Y, Z>{}, X{}, W{}};    // error: conflicting types deduced; operator Y not considered
```
Comment (burblebee) [inline source/overloading.tex]: use \tcode for code expressions in comments

File: source/overloading.tex (hunk 1 above)
Comment (jensmaurer) [reply source/overloading.tex]: Yes, but I don't want it. (This is a run-on sentence.)

File: source/overloading.tex (hunk 2 above)
Comment (jensmaurer) [reply source/overloading.tex]: Same as above.

File: source/overloading.tex (hunk 3 above)
Comment (jensmaurer) [reply source/overloading.tex]: Fixed.

File: source/overloading.tex (hunk 4 above)
Comment (jensmaurer) [reply source/overloading.tex]: Fixed.

**Pull request merged**